.github/workflows/android_tests.yml

```diff
@@ -0,0 +1,43 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Android tests
+
+# Controls when the action will run. 
+on:
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  test:
+    runs-on: macos-latest
```

Comment:
Do we need to run on a macos host? Perhaps `ubuntu-latest` would suffice? My assumption is that macOS hosts may be slower / more expensive, but not sure if that's true.